### ISSUE TITLE
Expose AnimatedSprite's 'play_backwards()' method as an inspector property

### ISSUE THREAD

**Issue by mathias17** [godotengine/godot-proposals#5286]
### Describe the project you are working on

(using Godot 3.5 stable)

I am creating a 2D game with a ton of decorative animations all over the environments that shouldn't require scripting because the majority of these animations are purely aesthetic.

### Describe the problem or limitation you are having in your project

As a non-scripter, I am unable to easily reverse the playback of an AnimatedSprite's animation.
As a result I have to use less convenient methods to achieve my goal.
___

Godot has been mostly great for the project's needs so far but as I add more and more decorative animations I am taking a liking to the simple convenience of the AnimatedSprite node because it does more than enough of what many of my simple looping animations require. I like that it doesn't require an AnimationPlayer in order to animate, like the default Sprite node does - cleaner scene trees to manage, less clicks, less time.

After getting into the habit of using AnimatedSprite lately, I believe our inability to reverse AnimatedSprite node animations without using a script is an easily mendable weakness of the node.
Allowing non-scripters to reverse AnimatedSprite animations would also make the node more consistent with the AnimationPlayer, which does provide a means of reversing animation in the inspector properties.

I'm not a scripter. I handle art and implementing art into the engine. If there's an elegant way to enable users like me to create the visual effects they want, without code, it seems consistent with Godot's user friendly engine style.

**AnimatedSprite inspector properties:**
![insp](https://user-images.githubusercontent.com/3425436/187123892-c6a89b09-f6f9-495c-854f-a1ce1e57da1d.png)

_Speed Scale does not accept negative values. If it did, there would be no problem._
I believe it makes perfect sense to allow a negative Speed Scale AND also have a play_backwards boolean. It may be an edge case to toggle play_backwards with a negative speed scale in effect but it's not an unreasonable thing to do in some circumstances.

___
**USE CASE**
![wateryshapeanim](https://user-images.githubusercontent.com/3425436/187126479-ecd21f22-3425-473d-9ab0-f676fcf4e650.gif)
Here you see me adding an animated watery shape to a pool of green liquid.
This "warbler" AnimatedSprite has a directional watery flow movement to its animation.
I want all watery animated elements to have a consistent rightward directional movement to them.
I can achieve more visual variations, using the same animation, by X-flipping and/or Y-flipping selective instances.
But, X-flipping results in the animation "going the other way". My solution for this problem is always to simply reverse the loop.
But I can't - not with an AnimatedSprite.

Yet, with only inspector properties, I _can_ reverse animations handled by an AnimationPlayer:
![animplayer](https://user-images.githubusercontent.com/3425436/187127567-223833a2-ffa8-4955-9559-defdff69f72b.png)
Setting "Speed" to -1 works great.

I find it strange and confusing that I can't do the same with an AnimatedSprite.

___

**Existing github posts:**
play_backwards was added in 2018, but only as a method: [https://github.com/godotengine/godot/pull/18044](https://github.com/godotengine/godot/pull/18044)
play_backwards as a property, being discussed: [https://github.com/godotengine/godot/pull/41821](https://github.com/godotengine/godot/pull/41821)

AnimatedSprite node docs: [https://docs.godotengine.org/en/stable/classes/class_animatedsprite.html](https://docs.godotengine.org/en/stable/classes/class_animatedsprite.html)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

In essence, I just want a way to reverse a AnimationSprite's playback.

It seems best to me, to try to match the AnimationPlayer node's functionality when it comes to playback features, so Godot's core animation ui is a little more consistent.

**Proposed changes:**
1)  Add the "play_backwards" boolean to the inspector properties for AnimatedSprite nodes.

2)  Allow negative values be input for an AnimatedSprite's "speed_scale" property.
(The AnimationPlayer node allows negative values and uses a slider control in the inspector.)

3)  Rename the inspector panel label for the AnimationPlayer node's "playback_speed" property from "Speed" to "Speed Scale".
This way both nodes use the same inspector property label for the same function.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I hope I covered this well enough above. Please see above.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I'm not a scripter so this is outside my wheelhouse, but the issue is more about user friendliness and convenience, therefore saving users time while they're working on common tasks.

### Is there a reason why this should be core and not an add-on in the asset library?

This proposal entails a minor unobtrusive change to a very commonly used node.
It's intended to potentially benefit anyone configuring animations in Godot.

**Comment by Mickeon** [comment]
See https://github.com/godotengine/godot-proposals/issues/4252 and https://github.com/godotengine/godot-proposals/issues/4506

**Comment by Calinou** [comment]
Shouldn't AnimatedSprite's speed scale property be allowed to be set to negative values, so you can use the same approach as AnimationPlayer?

**Comment by mathias17** [comment]
> Shouldn't AnimatedSprite's speed scale property be allowed to be set to negative values, so you can use the same approach as AnimationPlayer?

Absolutely, yes!
I would consider allowing negative Speed Scale values a sufficient solution.

I agree that access to a play_backwards property as well may be potentially confusing for some, but in my personal animation experience, being able to reverse an animation by ticking a box is powerful and useful.

I'm sure you guys have a better sense of what's best for Godot.
I'll be happy with any change that results in us non-scripters being able to reverse AnimatedSprite node animations without scripts or AnimationPlayer nodes.

**Comment by Mickeon** [comment]
My earlier proposals were first drafted as allowing negative numbers to play animations backwards, but I thought that having the speed scale jump between negative and positive numbers when `play()` is called with the second parameter would've been a tad unintuitive. In retrospective, however, the ability to "switch gears" that way is more useful and definitely easier than multiplying by -1 (and making sure the multiplication doesn't happen again, otherwise it'd be stuck switching rapidly between positive and negative `speed_scale`)

**Comment by mathias17** [comment]
I'm not sure I'm following you, @Mickeon. Are you for or against showing play_backwards in the inspector panel? English, man, English! 🐀 

There are two big thoughts on my mind here - 
1) I feel like AnimatedSprite's inspector properties and AnimationPlayer's "Playback" inspector properties should be as unified and similar as possible. This means the properties being very similar and the inspector panel property labels being consistent, etc.
(Of course it's a natural outcome for Godot's overall ui style/nomenclature to become a little piecemealed/inconsistent since so many people are working on it in tandem.)

2) This is out of left field, but . . . what if AnimatedSprite was deprecated - all its functionality merged into the default 2D Sprite node?
Ignoring the backwards compatibility woes, what if Sprite just did it all? Before improving AnimatedSprite, is the current consensus such that it's even necessary in the first place?
Sprite allows us to use an "AnimatedTexture" resource. What if it also allowed a "SpriteFrames" resource and had its own play controls?

**Comment by Mickeon** [comment]
> I'm not sure I'm following you, @Mickeon. Are you for or against showing play_backwards in the inspector panel?

Nowhere I stated my opinion on the matter in the comment, rather the opinion on the alternative.

I believe showing a `play_backwards`, `playing_backwards`, `backwards` property in the inspector to be needlessly confusing. At least, not nearly as intuitive as setting the speed to a negative value. Hence one of the reasons as to why I closed one of my proposals and reconsidered a PR.
